### PR TITLE
Add: infer primary columns from table definition 

### PR DESCRIPTION
**Pull request by kklein**

### CODE REVIEW

**Comment by kklein** [comment]
I took the liberty to pull from the fork. Up to you @YYYasin19  whether you want to use this as an actual PR moving forward or merely for seeing integration test results

**Comment by YYYasin19** [comment]
I overwrote the changes here with the other branch to clean it up and have the tests running 👍

**Comment by kklein** [comment]
Lgtm! Thank you! (I can't approve bc I created the PR)

File: CHANGELOG.rst [hunk 1]
```diff
@@ -15,7 +15,7 @@ Changelog
 - Implemented specification of number of counterexamples in :meth:`~datajudge.WithinRequirement.add_varchar_regex_constraint`.
 - Implemented in-database regex matching for some dialects via ``computation_in_db`` parameter in :meth:`~datajudge.WithinRequirement.add_varchar_regex_constraint`.
 - Added support for BigQuery backends.
-
+- Implemented an option ``infer_pk`` to automatically retrieve and check primary key constraints as part of the `datajudge.WithinRequirement.add_uniqueness_constraint`.
```
Comment (ivergara) [inline CHANGELOG.rst:18]: I think this should be moved to a new entry in the changelog.

**Review by ivergara — APPROVED** [review]

**Comment by YYYasin19** [comment]
(I have merged the other branch after validating that all of the same tests also passed here again)

**Pull request closed without merging**